macos/FluentUI/Button/Button.swift

```diff
@@ -253,6 +253,43 @@ open class Button: NSButton {
 		let path = NSBezierPath(roundedRect: bounds, xRadius: cornerRadius, yRadius: cornerRadius)
 		path.fill()
 	}
+	
+	/// Indicates if the Window that the button view has been added to, is inactive/backgrounded
+	private var isWindowInactive: Bool? {
+		didSet {
+			guard oldValue != isWindowInactive else {
+				return
+			}
+			// Re-compute the Button's color values for the latest Window State, and re-render it
+			setColorValues(forStyle: style, accentColor: accentColor, isWindowInactive: isWindowInactive ?? false)
+			needsDisplay = true
+		}
+	}
+
+	public override func viewDidMoveToWindow() {
```

Comment:
Should we update the value of isWindowInactive based on whether the window we're moving to is main?

Reply:
Sorry clarify? because NSWindow.didBecomeMainNotification handler already takes care of that below? by we're moving to instead of "we've moved to", are you talking about viewWillMoveToWindow?

Reply:
We register for notifications so we do something when the window's main-ness changes. What if the window is already main when the button is added?